src/detection.h

```diff
@@ -0,0 +1,4 @@
+#ifndef __DETECTION_H
+#define __DETECTION_H
+
+#endif /* __DETECTION_H */
```

Comment:
Should not we include here the `tim1_up_isr` header? In example:

```c
#ifndef __DETECTION_H
#define __DETECTION_H


void tim1_up_isr(void);


#endif /* __DETECTION_H */
```

Reply:
Done :)